src/app/new-auth/login/login.component.ts

```diff
@@ -1,23 +1,121 @@
-import { Component, OnInit } from '@angular/core';
+import { animate, style, transition, trigger } from '@angular/animations';
+import { Component, OnInit, ViewChild } from '@angular/core';
 import { NgForm } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
 import { AuthValidationService } from 'app/core/auth-validation.service';
+import { AuthService } from 'app/core/auth.service';
 
 @Component({
   selector: 'clark-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.scss']
+  styleUrls: ['./login.component.scss'],
+  animations: [
+    trigger(
+      'inOutAnimation',
+      [
+        transition(
+          ':enter',
+          [
+            style({ height: 0, opacity: 0 }),
+            animate('.5s ease-out',
+                    style({ height: 85, opacity: 1 }))
+          ]
+        ),
+        transition(
+              ':leave',
+              [
+                style({ height: 85, opacity: 1 }),
+                animate('.5s ease-in',
+                        style({ height: 0, opacity: 0 }))
+              ]
+            )
+          ]
+        )
+      ]
 })
 export class LoginComponent implements OnInit{
 
-  loginFailure = false;
-  constructor(public authValidation: AuthValidationService) { }
+  loginFailure: Boolean = false;
+  isNameLogin = false;
+  authInfo: {username: string, password: string};
+  errorMsg = '';
+  bannerMsg = '';
+  attempts = 0;
+  @ViewChild('username')
+  username;
+  @ViewChild('password')
+  password;
+  redirectUrl;
+  constructor(
+    private authValidation: AuthValidationService,
+    private route: ActivatedRoute,
+    private auth: AuthService,
+    private router: Router
+    ) {
+      this.route.parent.data.subscribe(() => {
+        if (route.snapshot.queryParams.redirectUrl) {
+          this.redirectUrl = decodeURIComponent(route.snapshot.queryParams.redirectUrl);
+        }
+      });
+     }
+
+  ngOnInit(): void {
+    this.authValidation.getErrorState().subscribe(err => this.loginFailure = err);
+  }
 
-  ngOnInit(): void { }
 /**
- * TO-DO: implement this method
+ * checks for 'required' field errors
+ * then logs the user in
  *
  * @param form data from NgForm
  */
-  public submit(form: NgForm): void { }
+  public submit(form: NgForm) {
+    this.authInfo = form.value;
+    //get form control for all fields
+    const pwordFormCtl = this.password.valueAccessor.control;
+    const usernameFormCtl = this.username.valueAccessor.control;
+
+    if(usernameFormCtl.hasError('required') || pwordFormCtl.hasError('required')) {
+      this.bannerMsg = 'Please fill out all required fields';
+      this.authValidation.showError();
+    } else if (this.attempts < 5){
+      this.attempts++;
```

Comment:
If someone leaves their page open for an hour to wait does this reset? I know it sounds crazy, but just wondering.

Reply:
thats a good point, it also isn't persistent if the browser is closed, like if someone maxes out their attempts, quits and tries again in 45min it'll be back at 0. I can figure out the first problem but the second one seems tricky if it can even be done on the client.

Reply:
That's a session implementation that we haven't gotten to on the backend.